internal/generator/config/curve.go

```diff
@@ -22,6 +23,29 @@ type Curve struct {
 	FpInfo, FrInfo Field
 	G1             Point
 	G2             Point
+
+	HashE1 *HashSuite
+}
+
+type Isogeny struct {
+
+	//Isogeny to original curve
+	XMap RationalPolynomial
+	YMap RationalPolynomial // The y map is also evaluated on x. The result is multiplied by y.
+}
+
+type RationalPolynomial struct {
+	NumHex []string
+	DenHex []string //Denominator is monic. The leading coefficient (1) is omitted.
+}
+
+type HashSuite struct {
+	AHex string
+	BHex string
+
+	Z int // z (or zeta) is a quadratic non-residue with //TODO: some extra nice properties, refer to WB19
+
+	Isogeny *Isogeny //pointer so it's nullable. TODO: Bad practice or ok?
```

Comment:
common practice (have pointers to have a "zero value" set to the nil pointer)

Reply:
Not sure I understood. So this _is_ common practice?

Reply:
yes